### PR TITLE
spirv-fuzz: Fix to fact manager

### LINKED ISSUE #3338: spirv-fuzz: Fact manager can mutate the constant manager when querying constant facts
The fact manager's "GetConstantId" method calls the constant manager's "GetConstant" method, which adds a constant to the constant pool if not already present.  This can lead to a constant that doesn't have a corresponding id in the module being added to the constant manager.

While this hasn't yet led to a functional error, we should maintain the invariant (in spirv-fuzz at least) that the constant manager only knows about constants that exist in the module.

@Vasniktel @paulthomson FYI.

### PR DESCRIPTION
**Pull request by afd**
Reworks the way uniforms with known constants are queried to avoid
unintended side effects to the constant manager.

Fixes #3338.

### CODE REVIEW

File: source/fuzz/fact_manager.cpp [hunk 1]
```diff
@@ -159,9 +159,26 @@ uint32_t FactManager::ConstantUniformFacts::GetConstantId(
...
+  const opt::analysis::Constant* known_constant;
+  if (type->AsInteger()) {
+    opt::analysis::IntConstant candidate_constant(
+        type->AsInteger(), GetConstantWords(constant_uniform_fact));
+    known_constant =
+        context->get_constant_mgr()->FindConstant(&candidate_constant);
+  } else {
+    assert(
+        type->AsFloat() &&
+        "Uniform constant facts are only supported for int and float types.");
```
Comment (Vasniktel) [inline source/fuzz/fact_manager.cpp:171]: I'm curious whether uniform variables that have types different from `int` or `float` can have `FactConstantUniform` applied.

Comment (Vasniktel) [reply source/fuzz/fact_manager.cpp:171]: Or why can't they in this case.

Comment (afd) [reply source/fuzz/fact_manager.cpp:171]: spirv-fuzz accepts facts of the form: "uniform + indices" == "integer or float constant".

If you have a uniform u that is a struct consisting of a vec4 and an int2, you could say that the vec4 component is equal to (0.0, 0.5, 1.0, 1.5) by asserting four facts:

u[0][0] == 0.0
u[0][1] == 0.5
u[0][2] == 1.0
u[0][3] == 1.5

As almost everything in SPIR-V boils down to float or int (with the exception of bool) pretty much anything can be expressed this way (albeit in a somewhat cumbersome manner).

Comment (Vasniktel) [reply source/fuzz/fact_manager.cpp:171]: Thanks, that explains it.

**Pull request merged**